templates/common/security-groups/DSMSGIngressRules.template

```diff
@@ -0,0 +1,179 @@
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMHeartbeatIngressFromELB:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMRelayIngressFromELB:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      SourceSecurityGroupId: !Ref ELBSourceSG
+  DSMAgentIngress:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4118'
+      ToPort: '4118'
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMConsoleIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMHeartbeatIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMRelayIngressSelf:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      SourceSecurityGroupId: !Ref DSMSG
+  DSMConsoleIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      CidrIp: 192.168.0.0/16
+  DSMHeartbeatIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      CidrIp: 192.168.0.0/16
+  DSMRelayIngress1921918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      CidrIp: 192.168.0.0/16
+  DSMConsoleIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPGUIPort
+      ToPort: !Ref DSIPGUIPort
+      CidrIp: 10.0.0.0/8
+  DSMHeartbeatIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: !Ref DSIPHeartbeatPort
+      ToPort: !Ref DSIPHeartbeatPort
+      CidrIp: 10.0.0.0/8
+  DSMRelayIngress101918:
+    Type: 'AWS::EC2::SecurityGroupIngress'
+    Properties:
+      GroupId: !Ref DSMSG
+      IpProtocol: tcp
+      FromPort: '4122'
+      ToPort: '4122'
+      CidrIp: 10.0.0.0/8
```

Comment:
We deliberately decided not to expose the CIDR ranges to the customer as most users will be confused by what to enter for the parameter. The expectation is the customer will have private instances located in the DSM VPC (across all subnets) that need to connect with the DSM and these rules allow those connections regardless of what the CIDR range of the VPC has.